Gems/Atom/Asset/Shader/Code/Source/Editor/ShaderBuildArgumentsManager.h

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/base.h>
+#include <AzCore/IO/Path/Path.h>
+
+#include "CommonFiles/CommonTypes.h"
+#include <Atom/RHI.Edit/ShaderBuildArguments.h> 
+
+
+namespace UnitTest
+{
+    class ShaderBuildArgumentsTests;
+}
+
+namespace AZ
+{
+    namespace ShaderBuilder
+    {
+        //! This class manages a stack of ShaderBuildArguments.
+        //! It simplifies command line argument definition at each level
+        //! of the shader build hierarchy:
+        //! 
+        //! {
+        //! push(global arguments)
+        //!     {
+        //!     push(PlatformInfo arguments)
+        //!         {
+        //!         push(RHI arguments)
+        //!             {
+        //!             push(.shader arguments)
+        //!                 {
+        //!                 push(shader.supervariant arguments)
+        //!                 ...
+        //!                 -- build shader with current arguments. --
+        //!                 ...
+        //!                 pop()
+        //!                 }
+        //!             pop() 
+        //!             }
+        //!         pop()
+        //!         }
+        //!     pop()
+        //!     }
+        //! pop()
+        //! }
+        //!
+        //! At each push(), two set of arguments are necessary, the "remove" set and the "add" set.
+        //! The idea is that it allows deep customization of all the shader build arguments by removing
+        //! or adding arguments at each level.
+        class ShaderBuildArgumentsManager final
```

Comment:
Can we get some unit tests for this class please?

Reply:
It is already there. See: 
 TEST_F(ShaderBuildArgumentsTests, InitializeShaderBuildArgumentsManager_ValidateAllOperations)